### DISCUSSION TITLE
UUID for Remote Endpoint API

### DISCUSSION THREAD

**Discussion by hbiyik** [Tribler/tribler#6622] [Q&A]
https://github.com/Tribler/tribler/blob/87465a0609b7d8ebda36fda4926908f61c86e49e/src/tribler-core/tribler_core/components/gigachannel/community/gigachannel_community.py#L204

isnt random uuid supposed to be received from request query string so that in case multiple queires are made, the client can identify which response belongs to which?

@ichorid

**Comment by ichorid** [comment]
Ok, to clarify this: UUIDs are big and inefficient, so, these are only used locally. Remote requests are tracked and distinguished by the `.number` attribute of the `SelectRequest` class:

https://github.com/Tribler/tribler/blob/87465a0609b7d8ebda36fda4926908f61c86e49e/src/tribler-core/tribler_core/components/metadata_store/remote_query_community/remote_query_community.py#L177

**Comment by hbiyik** [comment]
Oh, i think i did not express it clearly and caused misunderstanding.
By client I mean Rest Api client. When i issue a remote serach from rest api, i can not filter the events endpoint with requested uuid. Because the output is something like:

`data: {"type": "remote_query_results", "event": {"results": [], "uuid": "2474c432-8361-4ed8-b9fa-9418c81991d9", "peer": "9c993f2b5db22077172f65a9f98328e2146865d2"}}`

And the uuid is randomly created by the tribler_core on each rest request, instead i would exptect core to receive it from the rest clients query strings, if not available create random by itself ie:

`request_uuid = kwargs.get("uuid") or uuid.uuid4() `

so that rest client will know the remote qurey result will match his own uuid.

**Reply by ichorid** [reply]
Well, the usage of the remote query endpoint is as follows:
1. Create a remote search request by querying the `remote_query` endpoint.
2. Receive a UUID for the created search request (returned by `remote_query` enpoint), note it.
3. Wait for results to be returned over the Events endpoint (SSE mechanism).
4. Filter the results by UUID of the request you noted.


Now that I look into it, I noticed an inconsistency in our REST API: the request type must be POST, and _not_ PUT :confused:

**Comment by hbiyik** [comment]
Ok that would do, thanks for the answet
